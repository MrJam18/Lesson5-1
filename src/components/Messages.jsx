import { changeTimeFormat } from "./utils/changeTimeFormat";

const Messages = (props) => {
    console.log(props);

    return (
        <>
        <div className='messages'>
            {props.messages.map((message)=> message.text && 
            <div className={message.author !== props.chatAuthor ? 'message message-author-me' : 'message message-author-chatUser'} key={message.id}>
                <div className={"message-header"}> {message.author} <span className="message__time">{changeTimeFormat(message.id)}</span></div>
                {message.text}
            </div>
            )}
          </div>
          <div className="senderPlace" >
        </div>
        </>
    );
};

export default Messages;