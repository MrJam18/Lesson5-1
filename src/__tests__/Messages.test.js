import { render } from "@testing-library/react";
import Messages from "../components/Messages";

it('test Messages to snapshot', () => {
    const messages = [
    {text: 'hello World', id: 123, author: 'user'},
    {text: 'hello man', id: 456, author: 'bot'},
    {text: 'how are you?', id: 789, author: 'user'},
    ]
    const component = render(
      <Messages messages = {messages} />
    );
    expect(component).toMatchSnapshot();
})