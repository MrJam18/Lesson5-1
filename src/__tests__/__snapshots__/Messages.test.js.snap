// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Messages to snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="messages"
      >
        <div
          class="message message-author-me"
        >
          <div
            class="message-header"
          >
             
            user
             
            <span
              class="message__time"
            >
              04:00
            </span>
          </div>
          hello World
        </div>
        <div
          class="message message-author-me"
        >
          <div
            class="message-header"
          >
             
            bot
             
            <span
              class="message__time"
            >
              04:00
            </span>
          </div>
          hello man
        </div>
        <div
          class="message message-author-me"
        >
          <div
            class="message-header"
          >
             
            user
             
            <span
              class="message__time"
            >
              04:00
            </span>
          </div>
          how are you?
        </div>
      </div>
      <div
        class="senderPlace"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="messages"
    >
      <div
        class="message message-author-me"
      >
        <div
          class="message-header"
        >
           
          user
           
          <span
            class="message__time"
          >
            04:00
          </span>
        </div>
        hello World
      </div>
      <div
        class="message message-author-me"
      >
        <div
          class="message-header"
        >
           
          bot
           
          <span
            class="message__time"
          >
            04:00
          </span>
        </div>
        hello man
      </div>
      <div
        class="message message-author-me"
      >
        <div
          class="message-header"
        >
           
          user
           
          <span
            class="message__time"
          >
            04:00
          </span>
        </div>
        how are you?
      </div>
    </div>
    <div
      class="senderPlace"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
